pkg/store/proxy_heap.go

```diff
@@ -774,9 +756,9 @@ func newEagerRespSet(
 
 		// This should be used only for stores that does not support doing this on server side.
 		// See docs/proposals-accepted/20221129-avoid-global-sort.md for details.
-		if len(l.removeLabels) > 0 {
-			sortWithoutLabels(l.bufferedResponses, l.removeLabels)
-		}
+		// NOTE. Client is not guaranteed to give a sorted response when extLset is added
+		// Generally we need to resort here.
+		sortWithoutLabels(l.bufferedResponses, l.removeLabels)
```

Comment:
Hm, does it mean we will resort again in the querier? Maybe we want the strategy to be set in the querier and propagated to the store. This way we can decide in both places whether to resort or not.

Reply:
Querier should use lazyresponse set always right?

Reply:
As I understand it, if sorting strategy `querier`, then we need to use eager in the querier and stream from the store.

Reply:
Ah yeah but if we use querier sorting strategy we should resort here i think; otherwise we could use sorting strategy none right?

Reply:
Yes, so:
* sorting strategy querier = eager in querier, lazy in stores.
* sorting strategy store = eager in store, lazy in querier.

Reply:
yeah precisely ( but i would like to add querier later since its more complicated change )

Reply:
Agreed, we will need this but maybe for a separate PR.